python/test_MemberParser.py

```diff
@@ -148,12 +189,23 @@ def test_parse_invalid(self):
         r'uint_fast64_t disallowed // only allow fixed width integers with exact widths',
         r'std::int_least16_t disallowed // also adding a std namespace here does not make these allowed',
         r'std::uint_fast16_t disallowed // also adding a std namespace here does not make these allowed',
-        r'std::array<uint_fast16_t, 42> disallowedArray // arrays should not accept disallowed fixed width types'
+        r'std::array<uint_fast16_t, 42> disallowedArray // arrays should not accept disallowed fixed width types',
+
+        # Default values cannot be empty
+        r'int emptyDefault{} // valid c++, but we want an explicit default value here',
+
+        # Some cases which should be caught by the -fsyntax-only checks
+        r'std::array<int, 1> array{1, 2, 3} // too many values provided',
+        # Invalid user default value initialization
+        r'int weirdDefault{whatever, even space} // invalid c++ should not pass',
+        r'int initFromFloat{3.14f} // implicit conversions are not allowed'
```

Comment:
Examples of things we can actually catch during validation, because we have enough context from the member definition alone.